src/robot_state_publisher.cpp

```diff
@@ -160,13 +161,11 @@ void RobotStatePublisher::publishFixedTransforms(const std::string & tf_prefix,
   // loop over all fixed segments
   for (auto seg = segments_fixed_.begin(); seg != segments_fixed_.end(); seg++) {
     geometry_msgs::msg::TransformStamped tf_transform = kdlToTransform(seg->second.segment.pose(0));
-    std::chrono::nanoseconds now = std::chrono::high_resolution_clock::now().time_since_epoch();
+    rclcpp::Time now = clock_->now();
     if (!use_tf_static) {
-      now += std::chrono::milliseconds(500);  // 0.5sec in NS
+      now = now + rclcpp::Duration(0, 500000000);  // 0.5sec in NS
```

Comment:
[Duration has a constructor that uses std::chrono](https://github.com/ros2/rclcpp/blob/24080a458dd6cc227e29a97127209acca1ceb060/rclcpp/include/rclcpp/duration.hpp#L38)

This should work

```c++
rclcpp::Duration(std::chrono::milliseconds(500))
```

Reply:
Changed.